### ISSUE TITLE
Doesn't suppport more than 2 levels of folders

### ISSUE THREAD

**Issue by vynarim** [earlSt1/vtt-compendium-folders#116]
**Describe the bug**
I have a 3 or 4 levels folder structure in my compendium to organize items and actors but it seems it causes errors saving content: not immedialely but after reload of FVTT or compendium, a lot of items are outsides the folder they were placed in

**To Reproduce**
Create a 4 level deep folder structure
Put several items in it
Close FVTT and re open it
open the compendium

**Expected behavior**
Level of impricated folders should not be an issue

**Screenshots**
![image](https://user-images.githubusercontent.com/20367460/140613195-254f85c0-e1cd-4069-9a0c-1f7218d11035.png)

**Information**
- FoundryVTT Version : 0.8.9
- Compendium Folders Version : 2.3.53
- Game System + Version : Nephilime Legende 0.8.9
- All other modules disabled? (y/n) n

**Comment by earlSt1** [comment]
Thanks for the report. I've been working on improving this aspect of the module and will be able to share a testing version soon to see if it fixes it.

**Comment by earlSt1** [comment]
Hi, can you try using this testing version and see if it fixes it? You'll need to uninstall your current version of CF and install this one using this manifest link: https://raw.githubusercontent.com/earlSt1/vtt-compendium-folders/082-update/test/module.json

I strongly recommend backing up your compendiums/worlds just in case.

**Comment by vynarim** [comment]
Tests are ongoing, please just confirm this "test" version should appear as 2.3.53 (I uninstalled the current CF and installed the one from your Json link
![image](https://user-images.githubusercontent.com/20367460/141652705-ae1004ff-ace8-4bdc-a52c-2b54c54d6c82.png)

**Comment by earlSt1** [comment]
Yep it will look like the same version

**Comment by earlSt1** [comment]
Hi @vynarim any luck with testing?

**Comment by vynarim** [comment]
Hello @earlSt1 , 
Sorry for the delay, it was a crazy november for me

I dit some tests this morning but there is still an issue : after adding some items in my world and exporting them to the compendium, they were not added to the assiociated folder but to the root

![image](https://user-images.githubusercontent.com/20367460/144740576-d0df62d8-0885-4f96-a676-8ad287304589.png)

**Comment by vynarim** [comment]
Just saw you released a 2.3.63 version, does it includes the fix you did for testing supported folder levels?

**Comment by earlSt1** [comment]
Hi, I think I've finally found the cause of this issue. Can you try running the following command as a macro, then refreshing to see if it fixes it?
```js
game.settings.set('compendium-folders','cached-folder',{});
```

**Comment by dmaneyapanda** [comment]
I don't know if this is related, but I have been having intermittent trouble with new things that were in new folders created in the world appearing in the root when newly exported to the compendium using "Export Folder Structure", and the folders which they were supposed to go into not appearing there.  I thought it was a complicated problem due to hosting on the Forge, because I am not seeing the same problems in my local installation, but since I see the fix above deals with caching so perhaps it is related to that.

Essentially, the workaround fix I've discovered is to manually add a new folder in the compendium.  Doing this causes the folder structure to get rebuilt somehow, and then magically the missing folders appear and all the stuff which had been in root gets put into the new folders as they should be.

I'm happy to try the setting above, but not sure if it requires the testing version too, which is a bit harder for me to do on the Forge.

**Comment by earlSt1** [comment]
> 

Yep that sounds to me like a caching issue. I'm gonna try to find some time to fix it at the weekend. In the meantime that command above should make everything refresh properly (when you create a folder in the compendium, the cache is regenerated). You also dont need a testing version for it to work.

**Comment by vynarim** [comment]
> ```js
> game.settings.set('compendium-folders','cached-folder',{});
> ```

I launched the macro, nothing special happened and for the moment I didn't encounter the issue

![image](https://user-images.githubusercontent.com/20367460/151664934-af385a1b-3a53-4378-b9fa-cf7bea109e31.png)

**Comment by dmaneyapanda** [comment]
I'll also confirm that running the macro above seemed to fix the problem for me.
